src/main/java/io/cdap/plugin/kinesis/streaming/KinesisStreamingSource.java

```diff
@@ -59,6 +60,17 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
   }
 
+  @Override
+  public void prepareRun(StreamingSourceContext context) throws Exception {
+    super.prepareRun(context);
+
+    Schema schema = context.getInputSchema();
+    if (schema != null && schema.getFields() != null) {
+      recordLineage(context, config.referenceName, schema,
```

Comment:
Let's put this method in ReferenceStreamingSource instead of cdap

Reply:
so is it OK to duplicate the logic in different places?

Reply:
This ReferenceStreamingSource should be in hydrator-common like other reference classes, that way it is not duplicating the logic

Reply:
so should we move it now or later?

Reply:
we can just keep it here for now.